backend/app.py

```diff
@@ -183,64 +172,121 @@ def delete_node(hierarchy_id, node_id):
+
     # Create alternative
-    # Generate values with alternative_id
-        # Values must match up with measurement nodes
-        # Generate null values if a value is not provided for a measurement
+    alternative = Alternative(
+        hierarchy_id,
+        name=data["name"]
+    )
 
-    pass
+    if "values" in data:
+        for value in data["values"]:
+
+            # Generate null values if a value is not provided for a measurement
+            if "measure" in value:
+                measure = value["measure"]
+            else:
+                measure = None
+
+            if "local_value" in value:
+                local_value = value["local_value"]
+            else:
+                local_value = None
+                
+            if "global_value" in value:
+                global_value = value["global_value"]
+            else:
+                global_value = None
+
+            # Generate values with alternative_id
+            # Values must match up with measurement nodes
+            if value["nodeId"] in validNodeIds:
+                value = Value(value["nodeId"], measure, local_value, global_value)
+                alternative.values.append(value)
+            else:
+                abort(404, description="Node ID not valid")
+
+    db.session.add(alternative)
+    db.session.commit()
+
+    return alternative.to_dict(), 201
 
 
 # TODO: Get Alternative
 @app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>")
-def get_alternative(hierarchy_id, alternative_id):
+def get_alternative(hierarchy_id, alternative_id, methods=['GET']):
     hierarchy = Hierarchy.query.filter_by(id=hierarchy_id).first()
-    alternative = Node.query.filter_by(id=alternative_id).first()
+    if not hierarchy:
+        abort(404, description="Resource not found")
+
+    alternative = Alternative.query.filter_by(id=alternative_id).first()
+    if not alternative:
+        abort(404, description="Resource not found")
 
-    # Return alternative.to_dict()
+    return alternative.to_dict(), 200
 
 
 # TODO: Delete Alternative
-@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>")
+@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>", methods=['DELETE'])
 def delete_alternative(hierarchy_id, alternative_id):
-    pass
+    hierarchy = Hierarchy.query.filter_by(id=hierarchy_id).first()
+    if not hierarchy:
+        abort(404, description="Resource not found")
+
+    alternative = Alternative.query.filter_by(id=alternative_id).first()
+    if not alternative:
+        abort(404, description="Resource not found")
+
+    db.session.delete(alternative)
+    db.session.commit()
+
+    message = f"Alternative {alternative_id} Deleted"
+    return jsonify({"message": message}), 200
 
 
 # TODO: Patch Value
-@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>/value/<value_id>")
+@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>/value/<value_id>", methods=['PATCH'])
```

Comment:
When the frontend builds the alternative form it uses the hierarchy measurement nodes to map all inputs. So it already has the node_id handy, but substituting a new value_id would require more work and be messier to handle. Can you use node_id instead of value_id to update a value measure?